projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/RunJob.tsx

```diff
@@ -71,41 +71,139 @@ export default class RunJobDialog extends Component<IJobPathProp> {
-          showDialog({
-            title: 'Run Job',
-            body: <div className='vdk-run-dialog-message-container'>
-            <p className='vdk-run-dialog-message'>The job was executed successfully!</p>
-          </div>,
-            buttons: [Dialog.okButton()]
-          });
-        }
-        else{
-          message = "ERROR : " + message;
-          const  errorMessage = new VdkErrorMessage(message);
-          showDialog({
-            title: 'Run Job',
-            body: <div  className="vdk-run-error-message ">
+    if (status) {
+      showDialog({
+        title: 'Run Job',
+        body: (
+          <div className="vdk-run-dialog-message-container">
+            <p className="vdk-run-dialog-message">
+              The job was executed successfully!
+            </p>
+          </div>
+        ),
+        buttons: [Dialog.okButton()]
+      });
+    } else {
+      message = 'ERROR : ' + message;
+      const errorMessage = new VdkErrorMessage(message);
+      if (
+        !docManager ||
+        !(await handleErrorsProducedByNotebookCell(errorMessage, docManager))
+      ) {
+        showDialog({
+          title: 'Run Job',
+          body: (
+            <div className="vdk-run-error-message ">
               <p>{errorMessage.exception_message}</p>
               <p>{errorMessage.what_happened}</p>
               <p>{errorMessage.why_it_happened}</p>
               <p>{errorMessage.consequences}</p>
               <p>{errorMessage.countermeasures}</p>
-            </div>,
-            buttons: [Dialog.okButton()]
-          });
-        }
+            </div>
+          ),
+          buttons: [Dialog.okButton()]
+        });
+      }
+    }
   }
 }
+
+export const findFailingCellId = (message: String): string => {
+  const regex = /cell_id:([0-9a-fA-F-]+)/;
+  const match = message.match(regex);
+  if (match) return match[1];
+  return '';
+};
+
+/**
+ * Seperate handling for notebook errors - option for the user to navigate to the failing cell when error is produced
+ */
+export const handleErrorsProducedByNotebookCell = async (
+  message: VdkErrorMessage,
+  docManager: IDocumentManager
+): Promise<boolean> => {
+  const failingCellId = findFailingCellId(message.what_happened);
+  if (failingCellId) {
+    const { path: nbPath, cellIndex: failingCellIndex } = await getNotebookInfo(
+      failingCellId
+    );
+    if (nbPath) {
+      const navigateToFailingCell = async () => {
+        const notebook = docManager.openOrReveal(nbPath);
+        if (notebook) {
+          await notebook.revealed; // wait until the DOM elements are fully loaded
+          const children = Array.from(notebook.node.children!);
+          if (children) {
+            children.forEach(async element => {
+              if (element.classList.contains('jp-Notebook')) {
+                const cells = element.children;
+                const inx = Number(failingCellIndex);
+                if (inx < 0 || inx > cells.length) {
```

Comment:
Is it possible to get this kind of garbage number for the failing cell index? Why does it happen?

Reply:
@DeltaMichael the problem is that in the backend we use the notebook json to get the failingCellIndex while in the front end we use the DOM to get the cells there.
Since the loading of the DOM elements in the front end is not really synchronised this might lead to problems 
For example, cells.lenght can be 0 (tho it is really unlikely it is still possible) just because the rendering takes more time for one or another reason even though there are cells in the json schema 
I think I have fixed it with  await notebook.revealed but still want to have it in any case